opsdroid/connector/slack/connector.py

```diff
@@ -205,14 +205,23 @@ async def _get_channels(self):
             cursor = None
 
             while True:
-                channels = await self.slack_web_client.conversations_list(
-                    cursor=cursor, limit=self.channel_limit
-                )
-                self.known_channels.update({c["name"]: c for c in channels["channels"]})
-                cursor = channels["response_metadata"].get("next_cursor")
-
-                if not cursor:
-                    break
+                try:
+                    channels = await self.slack_web_client.conversations_list(
+                        cursor=cursor, limit=self.channel_limit
+                    )
+                    self.known_channels.update({c["name"]: c for c in channels["channels"]})
+                    cursor = channels["response_metadata"].get("next_cursor")
+
+                    if not cursor:
+                        break
+                except SlackApiError as error:
+                    if "ratelimited" in str(error):
+                        wait_time = float(error.response.headers["Retry-After"])
+                        _LOGGER.warning(_(f"Rate limit threshold reached. Retrying after {wait_time} seconds."))
+                        await asyncio.sleep(wait_time)
+                        continue
+                    else:
+                        raise
```

Comment:
We can use the same snippet above but throw a different exception and that should trigger this line

```python
@pytest.mark.asyncio
@pytest.mark.add_response(Exception)
async def test_search_history_messages_limit_more_than_1000(
    connector, mock_api, caplog
):
    caplog.set_level(logging.INFO)
    with pytest.raises(Exception):
        await connector.search_history_messages(
        "C01N639ECTY", "1512085930.000000", "1512085980.000000", 1001
    )
```